File: tests/nodejs/cases/case_193_closure_chain_async_10.js
/**
 * Task case 193: closure_chain_async_10 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case193ClosureChainAsync10(input) {
  const taskName = 'closure_chain_async_10';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const handler = () => payload.input;
  // ESCAPE: retained closure captures payload.
  retainedHandlers.push(handler);
  return 'ok';
}

module.exports = {
  case193ClosureChainAsync10,
};

File: tests/go/cases/case_193_closure_chain_async_10.go
package escape_tests

var retainedCase193 = []map[string]string{}

func Case193ClosureChainAsync10(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "closure_chain_async_10",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "closure_chain_async_10:" + raw,
	}
	handler := func() string { return payload["input"] }
	_ = handler
	// ESCAPE: closure payload persisted indirectly via retained metadata map.
	retainedCase193 = append(retainedCase193, map[string]string{"h": payload["input"]})
	return "ok"
}
